auth/gcloud/aio/auth/token.py

```diff
@@ -267,18 +326,158 @@ async def acquire_access_token(self, timeout: int = 10) -> None:
         else:
             raise Exception(f'unsupported token type {self.token_type}')
 
+        return resp
+
+
+class IapToken(BaseToken):
+    """An OpenID Connect ID token for a single IAP-secured service."""
+
+    default_token_ttl = 3600
+
+    def __init__(
+        self, app_uri: str,
+        service_file: Optional[Union[str, IO[AnyStr]]] = None,
+        session: Optional[Session] = None,
+        service_account: Optional[str] = None,
```

Comment:
noob question, why do we need both `service_file` and `service_account` here?

Reply:
Ah sorry I see:

```
`service_account`: an optional string denoting a GCP service account which
  takes the form of an email address. Only valid (and required!) for
  authentication with a project's authorized users. [Impersonating a service
  account][impersonating-sa] is required when generating an ID token in this
  case.
  ```

Which service account should be supplied here?

Reply:
nit: would renaming this to `impersonate_service_account` or similar be clearer?

Reply:
Funny you mention this, I originally had this argument named `impersonating_service_account` 😛